### ISSUE TITLE
Logits warper for batch generation

### ISSUE THREAD

**Issue by bnurbekov** [huggingface/transformers#14530]
# 🚀 Feature request
Is there a way to modify `logits_warper`, so that it will apply different parameters (`top_p`, `top_k`, `temperature`) for each item in a batch instead of applying the same parameters for all items in the batch:

https://github.com/huggingface/transformers/blob/f25a9332e8d091398ce96c462e02a467943c8eb9/src/transformers/generation_utils.py#L1562

## Motivation

It will very useful for inference because it will increase the throughput. Currently, you cannot really use #7552 for inference because the same parameters apply to the entire batch.

## Your contribution

Can help with implementing this feature and reviewing the PR.

**Comment by patrickvonplaten** [comment]
Hey @bnurbekov,

Sorry for replying so late. We recently allowed to pass customized logits wrapper to `generate()`. Could you maybe try to build a custom wrapper for your purpose this way? 

https://github.com/huggingface/transformers/pull/14779
